util/topgen/strong_random.py

```diff
@@ -45,6 +194,18 @@ def load(self, input_file):
                 else:
                     log.error("Read value not in range 0 - 255.")
                     sys.exit(1)
+        # Statistical tests operate on chunks of 20000 bits.
+        # In order to test the statistical quality of the loaded buffer,
+        # the buffer size must be a multiple of SEQ_SIZE_BYTES bytes.
+        if self.size() % self.SEQ_SIZE_BYTES:
+            log.error("The file size must be a multiple of 20000 bits.")
+            sys.exit(1)
+        # Testing statistical quality of the loaded buffer. The quality is
+        # sufficient only if all tests pass.
+        all_tests_pass = self.test_monobit() and self.test_poker() and \
+            self.test_runs() and self.test_long_run()
+        if not all_tests_pass:
+            log.error("Insuficient entropy in the buffer. Some test fail.")
```

Comment:
Wouldn't it be better to report which tests fail?

Reply:
+1

Reply:
Agreed
I've modified the error message to report exactly which tests fail.